gdal/ogrinfo.xml

```diff
@@ -0,0 +1,129 @@
+<tool id="ogrinfo" name="OGR Informations" version="@VERSION@">
+    <description>lists information about an OGR supported data source</description>
+    <macros>
+        <import>gdal_macros.xml</import>
+    </macros>   
+    <expand macro="gdal_requirements" />
+    <command detect_errors="exit_code"><![CDATA[
+        ogrinfo -ro '$input1'  -al 
+            #if $settings.advanced=='advanced'
+
+                #if $settings.condi_sql.sql=='sql':
+                    #if $settings.condi_sql.sql_statement!='':
+                        -sql '$settings.condi_sql.sql_statement'
+                    #end if
+                #end if
+               
+                $settings.so
+
+            #end if
+
+        > '$output'
+
+    ]]></command>
+    
+    <inputs>
+        <param type="data" name="input1" format="data" label="Gdal supported input file" help="http://www.gdal.org/ogr_formats.html"/>
+        <conditional name="settings">
+            <expand macro="gdal_advanced_params_select"/>
+            <when value="advanced">
+                <conditional name="condi_sql">
+                    <param name="sql" type="select" label="-sql statement"> 
+                        <option value="no_sql" selected="true">Don't use the -sql option</option>
+                        <option value="sql">Use the -sql option</option>
+                    </param>
+                    <when value="no_sql">
+                    </when>
+                    <when value="sql">
+                        <param name="sql_statement" type="text" label="Execute the indicated SQL"  help="-sql statement eg : SELECT * FROM layer WHERE X>10." value="">
+                            <sanitizer sanitize="False"/>
+                            <validator type="regex" message="Type a valid SQL query">^(?ims)\s*select\s+.*\s+from\s+.*$</validator>
+                        </param>
+                    </when>
+                </conditional>
+                <param name="so" label="Summary Only" help="-so" type="boolean" truevalue="-so" falsevalue="" checked="true"/>
+            </when>
+        </conditional>
+    </inputs>
+    
+    <outputs>
+        <data name="output" format="txt" label="OGR Info file on ${on_string}"/>
+    </outputs>
+    
+    <tests>
+        <test>
+            <param name="input1" value="ogrinfo_test.nc"/>
+            <param name="advanced" value="simple"/>
+            <output name="output">
+                <assert_contents>
+                    <has_text text="  NC_GLOBAL#institution=Puertos del Estado (PdE) - Mercator-Ocean (MO)"/>
+                    <has_text text="  NC_GLOBAL#longitude_max=2.499999046325684"/>
+                    <has_text text="Feature Count: 145"/>
+                </assert_contents>
+            </output>
+
+<!-- test ok if "planemo test ogrinfo" can fail if "planemo test" because of the input dataset name present in the output file change.
+            <output name="output" file="ogrinfo_test1_out.txt" lines_diff="6"/>
```

Comment:
was lines diff not working?

If you see this

```
- foo
+ bar
```

You need to have a lines-diff of 2.

Reply:
`lines_diff` is working, the issue was that dataset_X was wrote every line in the output. 
I changed line diff for an `assert_contents` `has_text` and it's ok now